lib/babe/crypto.go

```diff
@@ -84,6 +84,37 @@ func checkPrimaryThreshold(randomness Randomness,
 	return inoutUint.Compare(threshold) < 0, nil
 }
 
+func claimSecondarySlotVRF(randomness Randomness,
+	slot, epoch uint64,
+	authorities []types.Authority,
+	keypair *sr25519.Keypair,
+	authorityIndex uint32,
+) (*VrfOutputAndProof, error) {
+
+	secondarySlotAuthor, err := getSecondarySlotAuthor(slot, len(authorities), randomness)
+	if err != nil {
+		return nil, fmt.Errorf("cannot get secondary slot author: %w", err)
+	}
+
+	if authorityIndex == secondarySlotAuthor {
+		transcript := makeTranscript(randomness, slot, epoch)
+
+		out, proof, err := keypair.VrfSign(transcript)
+		if err != nil {
+			return nil, fmt.Errorf("cannot verify transcript: %w", err)
+		}
+
+		logger.Debugf("claimed secondary slot, for slot number: %d", slot)
+
+		return &VrfOutputAndProof{
+			output: out,
+			proof:  proof,
+		}, nil
+	}
+
+	return nil, errNotOurTurnToPropose
```

Comment:
nit flip the condition around to be `if authorityIndex != secondarySlotAuthor` to return early and have less indentation